File: arrays/contains-duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        HashSet<Integer> nonDups = new HashSet<>();
        for(int i=0;i<nums.length;i++) {
            if(nonDups.contains(nums[i]))
                return true;
            nonDups.add(nums[i]);
        }
            return false;
    }
}

File: arrays/Javascript/containsDuplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function (nums) {
  const numsLength = nums.length;
  const numsMap = {}
  for (let i = 0; i < numsLength; i++) {
    if (numsMap[nums[i]]) {
      return true;
    }
    numsMap[nums[i]] = 1;
  }
  return false;
};

// console.log(containsDuplicate([1, 2, 3, 1]));
